q2_taxa/tests/test_methods.py

```diff
@@ -99,3 +100,619 @@ def test_collapse_bad_level(self):
 
         with self.assertRaisesRegex(ValueError, 'of 0 is too low'):
             collapse(table, taxonomy, 0)
+
+
+class FilterTable(unittest.TestCase):
```

Comment:
Awesome tests!

Here's one more which should trigger the `_` behavior of `LIKE`:

Given these features:
```
k__g-foo;p__g-fighters
k__i-know;p__kung-foo
```
Using `include="k__g-foo", mode='contains'` should only match the first feature, but I think the current implementation will also match the second one (`__` will match to `un`).

Reply:
And then there's also the case of filtering down a feature-table to only 0-sum features, which will then result in all samples being removed. That should raise the same empty-table error as filtering all features.